extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -1253,6 +1253,96 @@ describe('Project: sdk style project content operations', function (): void {
 		should(project.files.find(f => f.relativePath === 'folder3\\')!).not.equal(undefined, 'folder3\\ should be loaded');
 		should(project.files.find(f => f.relativePath === 'folder3\\innerFolder\\')!).not.equal(undefined, 'folder3\\innerFolder\\ should be loaded');
 	});
+
+	it('Should handle deleting empty folders', async function (): Promise<void> {
+		const testFolderPath = await testUtils.generateTestFolderPath();
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.newSdkStyleProjectSdkNodeBaseline, testFolderPath);
+
+		const project: Project = await Project.openProject(projFilePath);
+
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(0);
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(0);
+
+		// add an empty folder
+		await project.addFolderItem('folder1');
+
+		// verify folder was added
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(1);
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(0);
+		should(project.files.find(f => f.relativePath === 'folder1\\')).not.equal(undefined, 'folder1 should have been added');
+
+		// verify entry was added for the new empty folder in the sqlproj
+		let projFileText = (await fs.readFile(projFilePath)).toString();
+		should(projFileText.includes('<Folder Include="folder1\\" />')).equal(true, projFileText);
+
+		// delete the empty folder
+		await project.deleteFileFolder(project.files.find(f => f.relativePath === 'folder1\\')!);
+
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(0, 'folder1 should have been deleted');
+
+		// verify the folder entry was removed from the sqlproj and a Build Remove was not added
+		projFileText = (await fs.readFile(projFilePath)).toString();
+		should(projFileText.includes('<Folder Include="folder1\\" />')).equal(false, projFileText);
+		should(projFileText.includes('<Build Remove="folder1\\**" />')).equal(false, projFileText);
```

Comment:
What about instead comparing against the original sqlproj file contents?  Confirming that a large string _doesn't_ contain a rather specific substring seems like it could yield false positives if formatting slightly changes at all, so a positive test rather than negative test would be more confident.